src/cluster_legacy.c

```diff
@@ -2051,6 +2086,14 @@ void clusterProcessGossipSection(clusterMsg *hdr, clusterLink *link) {
     clusterMsgDataGossip *g = (clusterMsgDataGossip*) hdr->data.ping.gossip;
     clusterNode *sender = link->node ? link->node : clusterLookupNode(hdr->sender, CLUSTER_NAMELEN);
 
+    /* Abort if the gossip contains invalid node IDs to avoid adding incorrect information to
+     * the nodes dictionary. An invalid ID indicates memory corruption on the sender side. */
+    int invalid_ids = verifyGossipSectionNodeIds(g, count);
+    if (invalid_ids) {
+        serverLog(LL_WARNING, "Node %.40s (%s) gossiped %d nodes with invalid IDs.", sender->name, sender->human_nodename, invalid_ids);
```

Comment:
I realized something after merging this. Sender could be NULL if the node is not known, one of the reasons this could happen is the sender is malicious, so we need to need to add NULL checks around sender here.

Reply:
Ah good catch. I didn't keep this code wrapped in a sender check when I moved it.

Reply:
Was fixed in https://github.com/redis/redis/pull/12980.